core/src/main/java/com/linecorp/armeria/client/DefaultClientFactory.java

```diff
@@ -171,4 +173,19 @@ public void close() {
     void doClose() {
         clientFactoriesToClose.forEach(ClientFactory::close);
     }
+
+    private URI normalizeUri(URI uri, Scheme scheme) {
+        if (uri == HttpClientBuilder.UNDEFINED_URI) {
+            // See https://github.com/line/armeria/pull/2241 discussion for more details.
```

Comment:
👍

Reply:
The PR is pretty long and I don't think it's practical to reference it to find this detail. We can just inline the explanation

```
// We use a special singleton marker URI for clients that do not explicitly define a host or scheme at construction time.
// As this isn't created by users, we don't need to normalize it.
```

Reply:
Done